src/aiida_quantumespresso/workflows/xspectra/crystal.py

```diff
@@ -0,0 +1,676 @@
+        builder.core.xs_prod.kpoints_distance = orm.Float(core_xspectra['kpoints_distance'])
+        builder.core.get_powder_spectrum = orm.Bool(True)
+        builder.core.abs_atom_marker = abs_atom_marker
+        core_hole_pseudos = {}
+        gipaw_pseudos = {}
+        if elements_list:
+            elements_not_present = []
+            elements_present = [kind.symbol for kind in structure.kinds]
+            for element in elements_list:
+                if element not in elements_present:
+                    elements_not_present.append(element)
+            if len(elements_not_present) > 0:
+                raise ValueError(
+                    f'The following elements: {elements_not_present} are not present in the'
+                    f' structure ({elements_present}) provided.'
+                    )
+            else:
+                builder.elements_list = orm.List(elements_list)
+                for element in pseudos:
+                    core_hole_pseudos[element] = pseudos[element]['core_hole']
+                    gipaw_pseudos[element] = pseudos[element]['gipaw']
+        # if no elements list is given, we instead initalise the pseudos dict with all
+        # elements in the structure
+        else:
+            for element in pseudos:
+                core_hole_pseudos[element] = pseudos[element]['core_hole']
+                gipaw_pseudos[element] = pseudos[element]['gipaw']
+        builder.core_hole_pseudos = core_hole_pseudos
+        builder.gipaw_pseudos = gipaw_pseudos
+        if core_hole_treatments:
+            builder.core_hole_treatments = orm.Dict(dict=core_hole_treatments)
+        if core_wfc_data:
+            builder.core_wfc_data = core_wfc_data
+        elif upf2plotcore_code:
+            builder.upf2plotcore_code = upf2plotcore_code
+        else:
+            raise ValueError(
+                'No code node for upf2plotcore.sh or core wavefunction data were provided.'
+            )
+        # pylint: enable=no-member
+        return builder
+
+
+    @staticmethod
+    def validate_inputs(inputs, _):
+        """Validate the inputs before launching the WorkChain."""
+        structure = inputs['structure']
+        kinds_present = [kind.name for kind in structure.kinds]
+        elements_present = [kind.symbol for kind in structure.kinds]
+        if 'elements_list' in inputs:
+            absorbing_elements_list = sorted(inputs['elements_list'])
+            extra_elements = []
+            for element in absorbing_elements_list:
+                if element not in elements_present:
+                    extra_elements.append(element)
+            if len(extra_elements) > 0:
+                raise ValidationError(
+                    f'The some elements in ``elements_list`` {extra_elements} do not exist in the'
+                    f' structure provided {elements_present}.'
+                )
+        else:
+            absorbing_elements_list = sorted([kind.symbol for kind in structure.kinds])
```

Comment:
We can actually just get rid of the line entirely and change `elements_present` on line 374 to be `sorted([kind.symbol for kind in structure.kinds])`, since it can then be re-used in the other checks if needed.